api/proto/config/storage.proto

```diff
@@ -106,3 +106,48 @@ message Drive {
   // A value of 0 will indicate that no resizing is required
   int64 maxsizebytes = 10;
 }
+
+
+message ContentTree {
+  UUIDandVersion uuidandversion = 1;
+  string dsId  = 2; // effectively pointer/key into dsConfigs
+
+  string URL = 3;
+  Format iformat = 4;
+
+  // the following is only used for individual blobs, if this message references a group of blobs
+  // e.g. in the case of OCI -- this information is expected to be provided by a top level blob
+  // that this message points to via its URL
+  string sha256 = 5;
+  uint64 sizeBytes = 6; // used for capping resource consumption eve for OCI & BLOB_TREE
+  SignatureInfo siginfo = 7;
+}
+
+enum VolumeAccessProtocols {
+  VAP_NONE = 0;
+  VAP_9P = 1;
+  VAP_NFS = 2;
+  VAP_SCSI = 3;
+  VAP_VIRTIO = 4;
+}
+
+message Volume {
+  UUIDandVersion uuidandversion = 1;
```

Comment:
Would be good to spec out what can change in a Volume config? And will the version be increased as the volume config changes?

Reply:
What would be the expectations for a volume change? 

1) Would the existing Apps supposed to see the changes immediately? Or would they require a purge?
2) I assume the following fields cannot change once a volume is created:
a) ContentTreeId
b) DriveType
c) voltype
d) readonly <--- Or can this change??

4) And the following can change:
a) protocols
b) maxSizeBytes

Based on this, We can decide:
1) Does the App Need a Purge to handle the change? etc..

We can start off by saying a Volume cannot be modified once it is created - or specify the fields that will not be changed. These can be put as comments..

Reply:
I don't think we need the version here. Just the UUID.

Reply:
I changed it to uuid.